tools/mason/MasonInit.chpl

```diff
@@ -72,16 +79,65 @@ proc masonInit(args) throws {
     }
 
     if dirName == '' {
-      const cwd = getEnv("PWD");
-      var name = basename(cwd);
-      const path = '.';
-      if packageName.size > 0 then name = packageName;
-      var resName = validatePackageNameChecks(path, name);
-      name = resName;
-      validateMasonFile(path, name, show);
-      var isInitialized = validateInit(path, name, true, show);
-      if isInitialized > 0 then
-      writeln("Initialized new library project: " + basename(cwd));
+      if defaultBehavior {
+        const cwd = getEnv("PWD");
+        var name = basename(cwd);
+        const path = '.';
+        if packageName.size > 0 then name = packageName;
+        var resName = validatePackageNameChecks(path, name);
+        name = resName;
+        validateMasonFile(path, name, show);
+        var isInitialized = validateInit(path, name, true, show);
+        if isInitialized > 0 then
+        writeln("Initialized new library project: " + basename(cwd));
+      } else {
+        // check if Mason.toml file and src/moduleFile is present
+        var isMasonTomlPresent = false;
+        var isSrcPresent = false;
+        var moduleName: string;
+        if isFile('./Mason.toml') then isMasonTomlPresent = true;
+        if isDir('./src') then isSrcPresent = true;
+        // parse values from TOML File && module file
+        var defaultPackageName: string;
+        var defaultVersion: string;
+        var defaultChplVersion: string;
+        if isMasonTomlPresent {
+          const toParse = open("./Mason.toml", iomode.r);
+          const tomlFile = owned.create(parseToml(toParse));
+          if tomlFile.pathExists("brick.name") then
+            defaultPackageName = tomlFile["brick"]!["name"]!.s;
+          if tomlFile.pathExists("brick.version") then
+            defaultVersion = tomlFile["brick"]!["version"]!.s;
+          if tomlFile.pathExists("brick.chplVersion") then
+            defaultChplVersion = tomlFile["brick"]!["chplVersion"]!.s;
+        }
+        if isSrcPresent {
+          var file: string = listdir("./src");
+          moduleName = file;
+        }
+       // begin interactive session and get values input by user
+        var newPackageName, newVersion, newChplVersion: string;
+        var result = beginInteractiveSession(defaultPackageName, defaultVersion, defaultChplVersion);
+        newPackageName = result[0];
+        newVersion = result[1];
+        newChplVersion = result[2];
+        // overwrite Toml File
+        validateMasonFile('.', newPackageName, show);
+        isMasonTomlPresent = true;
+        if isMasonTomlPresent {
```

Comment:
I would read the TOML file in once, update the values by modifying it in each of the if blocks, then write the TOML object to file, rather than read it in every time.